internal/store/batchdeleteeviction.go

```diff
@@ -0,0 +1,102 @@
+package store
+
+import (
+	"container/heap"
+
+	"github.com/dicedb/dice/internal/object"
+)
+
+// evictionItemHeap is a max-heap of evictionItems based on lastAccessed.
+type evictionItemHeap []evictionItem
+
+func (h *evictionItemHeap) Len() int { return len(*h) }
+
+func (h *evictionItemHeap) Less(i, j int) bool {
+	// For a max-heap, we want higher lastAccessed at the top.
+	return (*h)[i].lastAccessed > (*h)[j].lastAccessed
+}
+
+func (h *evictionItemHeap) Swap(i, j int) { (*h)[i], (*h)[j] = (*h)[j], (*h)[i] }
+
+func (h *evictionItemHeap) Push(x interface{}) {
+	*h = append(*h, x.(evictionItem))
+}
+
+func (h *evictionItemHeap) Pop() interface{} {
+	old := *h
+	n := len(old)
+	x := old[n-1]
+	*h = old[:n-1]
+	return x
+}
+
+// Encapsulate heap operations to avoid interface{} in main code.
+func (h *evictionItemHeap) push(item evictionItem) {
+	heap.Push(h, item)
+}
+
+func (h *evictionItemHeap) pop() evictionItem {
+	return heap.Pop(h).(evictionItem)
+}
+
+// BatchDeleteEvictionLRU implements batch eviction of least recently used keys
+type BatchDeleteEvictionLRU struct {
+	BaseEvictionStrategy
+	maxKeys     int
+	bulkPercent float64
+}
+
+func NewBatchDeleteEvictionLRU(maxKeys int, bulkPercent float64) *BatchDeleteEvictionLRU {
+	if bulkPercent <= 0 || bulkPercent >= 1 {
+		bulkPercent = 0.1 // default to 10% if invalid value provided
+	}
+	return &BatchDeleteEvictionLRU{
+		maxKeys:     maxKeys,
+		bulkPercent: bulkPercent,
+	}
+}
+
+func (e *BatchDeleteEvictionLRU) ShouldEvict(store *Store) int {
+	currentKeys := store.GetKeyCount()
+	if currentKeys < e.maxKeys {
+		return 0 // No eviction needed
+	}
+
+	// Calculate number of keys to evict
+	toEvict := int(float64(currentKeys) * (e.bulkPercent / 100))
+	if toEvict < 1 {
+		toEvict = 1 // Ensure we evict at least one key
+	}
+
+	return toEvict
+}
+
+// EvictVictims deletes keys with the lowest LastAccessedAt values from the store.
+func (e *BatchDeleteEvictionLRU) EvictVictims(store *Store, toEvict int) {
+	h := make(evictionItemHeap, 0, toEvict)
+	heap.Init(&h)
+
+	store.GetStore().All(func(k string, obj *object.Obj) bool {
+		item := evictionItem{
+			key:          k,
+			lastAccessed: obj.LastAccessedAt,
+		}
+		if h.Len() < toEvict {
+			h.push(item)
+			return true
+		}
+
+		if item.lastAccessed < h[0].lastAccessed {
+			h.pop()
+			h.push(item)
+		}
+		return true
+	})
+
+	for h.Len() > 0 {
+		item := h.pop()
+		store.Del(item.key, WithDelCmd(Evict))
```

Comment:
@JyotinderSingh wanted to discuss if we need some kind of special handling during eviction for QWATCH and WATCH implementations.